### PR TITLE
Expose ProtoToolchainInfo to Starlark

### PR DESCRIPTION
**Pull request by Yannic**
Progress on https://docs.google.com/document/d/1go3UMwm2nM8JHhI3MZrtyoFEy3BYg5omGqQmOwcjmNE/edit

### CODE REVIEW

**Comment by Yannic** [comment]
@katre PTAL

File: src/main/java/com/google/devtools/build/lib/starlarkbuildapi/proto/ProtoBootstrap.java [hunk 1]
```diff
@@ -28,23 +28,23 @@
...
 
   private final ProtoInfoProviderApi protoInfoApiProvider;
+  private final ProtoToolchainInfoApi.Provider protoToolchainInfoApi;
   private final Object protoCommon;
   private final StarlarkAspectApi protoRegistryAspect;
   private final ProviderApi protoRegistryProvider;
 
   public ProtoBootstrap(
       ProtoInfoProviderApi protoInfoApiProvider,
+      ProtoToolchainInfoApi.Provider protoToolchainInfoApi,
```
Comment (katre) [inline src/main/java/com/google/devtools/build/lib/starlarkbuildapi/proto/ProtoBootstrap.java:42]: ProtoInfo is passing the actual Api class, not the Api.Provider. Why the difference here?

**Review by katre — COMMENTED** [review]
Looks good, only one question I'm curious about.

File: src/main/java/com/google/devtools/build/lib/starlarkbuildapi/proto/ProtoBootstrap.java (hunk 1 above)
Comment (Yannic) [reply src/main/java/com/google/devtools/build/lib/starlarkbuildapi/proto/ProtoBootstrap.java:42]: `ProtoInfo` also passes the provider, but they named it `ProtoInfoApi.ProtoInfoProviderApi`

Comment (katre) [reply src/main/java/com/google/devtools/build/lib/starlarkbuildapi/proto/ProtoBootstrap.java:42]: Ah, I see. Thanks for clarifying.

**Review by katre — APPROVED** [review]
I'll merge this today.

**Comment by katre** [comment]
Okay, apparently our internal Java toolchain is stricter than what we have externally. Can you investigate and fix these warning/errors:

```
src/main/java/com/google/devtools/build/lib/starlarkbuildapi/proto/ProtoBootstrap.java:42: warning: [rawtypes] found raw type: Provider
      ProtoToolchainInfoApi.Provider protoToolchainInfoApi,
                           ^
  missing type arguments for generic class Provider<FilesToRunProviderApiT>
  where FilesToRunProviderApiT is a type-variable:
    FilesToRunProviderApiT extends FilesToRunProviderApi<? extends FileApi> declared in interface Provider
error: warnings found and -Werror specified
Target //src/main/java/com/google/devtools/build/lib/starlarkbuildapi/proto:proto failed to build
```

**Comment by Yannic** [comment]
cannot repro locally, but I added the missing type parameters. Let me know if there are more errors

**Pull request closed without merging**